File: common/src/main/java/com/pluxity/permission/dto/PermissionRequest.java
package com.pluxity.permission.dto;

import com.pluxity.permission.ResourceType;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotNull;
import java.util.List;

public record PermissionRequest(
        @NotNull
                @Schema(description = "자원 유형", implementation = ResourceType.class, example = "FACILITY")
                String resourceType,
        @NotNull @Schema(description = "자원 아이디") List<String> resourceIds) {}

File: common/src/main/kotlin/com/pluxity/permission/dto/PermissionRequest.kt
package com.pluxity.permission.dto

import com.pluxity.permission.PermissionLevel
import com.pluxity.permission.ResourceType
import io.swagger.v3.oas.annotations.media.Schema

data class PermissionRequest(
    @field:Schema(
        description = "자원 유형",
        implementation = ResourceType::class,
        example = "FACILITY",
    ) val resourceType: String,
    @field:Schema(
        description = "자원 아이디",
    ) val resourceIds: List<String>,
    @field:Schema(
        description = "권한 레벨",
        implementation = PermissionLevel::class,
        example = "READ",
    ) val level: PermissionLevel = PermissionLevel.READ,
)
